cognite/client/_api/datapoints.py

```diff
@@ -322,6 +322,55 @@ def insert_multiple(self, datapoints: List[Dict[str, Union[str, int, List]]]) ->
         dps_poster = DatapointsPoster(self)
         dps_poster.insert(datapoints)
 
+    @overload
+    def delete_point(self, point: Dict[str, Union[int, str, datetime]]) -> None:
+        ...
+
+    @overload
+    def delete_point(self, point: Sequence[Dict[str, Union[int, str, datetime]]]) -> None:
+        ...
+
+    def delete_point(
+        self, point: Union[Dict[str, Union[int, str, datetime]], Sequence[Dict[str, Union[int, str, datetime]]]]
+    ) -> None:
+        """Delete a single datapoint from one or more time series.
+
+        Args:
+            point (Union[Dict[str, Union[int, str, datetime]], Sequence[Dict[str, Union[int, str, datetime]]]]): The point or points to delete
+
+        Returns:
+            None
+
+        Examples:
+
+            Deleting a single datapoint from a single time series::
+
+                >>> from cognite.client import CogniteClient
+                >>> c = CogniteClient()
+                >>> c.datapoints.delete_point(point={"id": 1, "timestamp": 12345})
+
+            Deleting a single datapoint from multiple time series::
+
+                >>> from cognite.client import CogniteClient
+                >>> c = CogniteClient()
+                >>> c.datapoints.delete_point(point=[{"id": 1, "timestamp": 123}, {"externalId": "abc", "timestamp": datetime(year=1992)}])
+        """
+        point_list = point if isinstance(point, Sequence) else [point]
+        valid_ranges = []
+        for point in point_list:
+            for key in point:
+                if key not in ("id", "externalId", "timestamp"):
+                    raise AssertionError(
+                        f"Invalid key '{key}' in point. Must contain 'timestamp', and 'id' or 'externalId"
+                    )
```

Comment:
I think I've seen this type of logic in multiple places in the SDK, could be split out to utils.

Reply:
Will do 👍